src/core/DY.Core.Printing.fst

```diff
@@ -0,0 +1,113 @@
+module DY.Core.Printing
+
+open DY.Core.Trace.Type
+open DY.Core.Label.Type
+open DY.Core.Bytes.Type
+
+/// This module provides functions to convert types
+/// from DY* to a string. These functions are used to
+/// print the trace from an example protocol run.
+
+val label_to_string: (l:label) -> string
+let rec label_to_string l =
+  match l with
+  | Secret -> "Secret"
+  | State pre_label -> (
+    match pre_label with
+    | P p -> Printf.sprintf "Principal %s" p
+    | S p s -> Printf.sprintf "Principal %s state %d" p s
+  ) 
+  | Meet l1 l2 -> Printf.sprintf "Meet [%s; %s]" (label_to_string l1) (label_to_string l2)
+  | Join l1 l2 -> Printf.sprintf "Join [%s; %s]" (label_to_string l1) (label_to_string l2)
+  | Public -> "Public"
+
+val usage_to_string: (u:usage) -> string
+let usage_to_string u =
+  match u with
+  | SigKey tag -> "SigKey " ^ tag
+  | SigNonce -> "SigNonce"
+  | PkdecKey tag -> "PkdecKey " ^ tag
+  | PkNonce -> "PkNonce"
+  | AeadKey tag -> "AeadKey " ^ tag
+  | DhKey tag -> "DhKey " ^ tag
+  | NoUsage -> "NoUsage"
+
+val uint_list_to_string: list FStar.UInt8.t -> string
+let rec uint_list_to_string seq =
+  match seq with
+  | [] -> ""
+  | hd :: tl -> (
+    // Only convert certain ASCII characters to create clean strings
+    // without special characters.
+    if FStar.UInt8.gt (FStar.UInt8.uint_to_t 32) hd || FStar.UInt8.lt (FStar.UInt8.uint_to_t 122) hd then uint_list_to_string tl
+    else Printf.sprintf "%c%s" (FStar.Char.char_of_int (FStar.UInt8.v hd)) (uint_list_to_string tl)
+  )
+
+val bytes_to_string: (b:bytes) -> string
```

Comment:
In this function please add a newline for multiline match case and only one level of indentation, as you have done in `trace_event_to_string`